tests/wpunit/RequestTest.php

```diff
@@ -88,18 +88,6 @@ public function testRequestError() {
 		$this->assertEquals( 1, count( $results['errors'] ) );
 	}
 
-	/**
-	 * The request should not clobber the global post since it can be called in the Loop.
-	 */
-	public function testRequestPreservesGlobalPost() {
-		$GLOBALS['post'] = 'testing';
-
-		$this->factory->post->create();
-		$this->create_example_request()->execute();
-
-		$this->assertEquals( 'testing', $GLOBALS['post'] );
-	}
```

Comment:
Seems like it would be good to keep this test? Why delete it?

Reply:
I originally removed it because it wasn't working with my implementation because it does exercise a weird use case: Setting global post to a string but not calling `setup_postdata` which doesn't make any sense? 

The current implementation in this PR passes it now but due to its weirdness and because it seems that testPostQueryPostDataReset already tests this situation well enough I decided to leave it removed. 

https://github.com/wp-graphql/wp-graphql/blob/c9ca28a1d1bdcf34a773220e28412b401dc4e907/tests/wpunit/PostObjectQueriesTest.php#L1321

But if you think there's some merit to it I'll restore it.

Reply:
@epeli Ya, the use for this test was to make sure _whatever_ is set as the Global post would be the same _after_ a GraphQL request was made. 

Since you can use a GraphQL Query within the context of a post, such as within a Shortcode or whatever, we want to make sure that whatever the Global post is _before_ GraphQL execution is the same as it is _after_ GraphQL execution. In this case, we set the global to `testing` and assert that it's the same _after_ GraphQL executes. 

I believe at one point we weren't properly resetting, so if you used GraphQL within a shortcode, anything on the page after that (such as related posts after the conent, etc) would think the Global Post was the last item resolved by GraphQL. 

This test _should_ prevent that regression.

Reply:
Fair enough. The test will indeed ensure wp-graphql stays resilient even if the user does something wonky with the global post. I'll restore it now.

Reply:
ya, we don't really care _what_ the global post was set to before a GraphQL request, we just want to ensure it's the same before and after. 😄 Thanks for fixing!